packages/shared/blockchain/evm-based/src/lib/evm-based.utils.ts

```diff
@@ -96,7 +96,7 @@ export const evmBasedUtils = {
       estimate = new BigNumber(await client.eth.estimateGas(txWithoutGas))
     } catch (e) {
       if (gas) estimate = new BigNumber(gas as string)
-      else throw new EvmBasedSdkError({ error: e as Error })
+      else throw new EvmBasedSdkError({ error: new Error(`Unable to estimate gas, reason ${e}`) })
```

Comment:
not like this, check how we do it e.g. here - https://github.com/tatumio/tatum-js/blob/master/packages/blockchain/xrp/src/lib/services/xrp.tx.ts#L108

Reply:
This error is only returned when it fails to calculate gas for some reasons, maybe there is an issue with the transaction data for similar error as shown in the [link]( https://github.com/tatumio/tatum-js/pull/617/files#diff-e0977f8da3d51fe35030e19de0961c735cece95d55bdd64e0c913aeaed337f86R107-R112). What do you think?

Reply:
I am just saying how the format of thrown error should look like.